primitives/parachain-inherent/src/client_side.rs

```diff
@@ -104,6 +104,7 @@ async fn collect_relay_storage_proof(
 	relevant_keys.push(relay_well_known_keys::hrmp_egress_channel_index(para_id));
 	relevant_keys.push(relay_well_known_keys::upgrade_go_ahead_signal(para_id));
 	relevant_keys.push(relay_well_known_keys::upgrade_restriction_signal(para_id));
+	relevant_keys.push(relay_well_known_keys::para_head(para_id));
```

Comment:
@bkchr do you think it's an issue if we merge this into master and proofs start carrying the `para_head` in the proof even if it's unused?

Alternatively, we could refactor this bit of code and create a new version of the `ParachainInherentData`. The old version would not have this member, and the new version would.

Reply:
There's feature branch for cumulus if we don't want this hustle